Index: config/global.ini.php
===================================================================
--- config/global.ini.php	(revision 4837)
+++ config/global.ini.php	(working copy)
@@ -428,6 +428,9 @@
 Plugins[] 		= UserCountryMap
 Plugins[] 		= Live
 Plugins[]		= CustomVariables
+Plugins[]		= ThoughtFarmer
+Plugins[]		= ThoughtFarmerUser
+
 [PluginsInstalled]
 PluginsInstalled[] = Login
 PluginsInstalled[] = CoreAdminHome
