--- core/ArchiveProcessing/Period.php	(revision 5170)
+++ core/ArchiveProcessing/Period.php	(working copy)
@@ -373,7 +373,8 @@
 							FROM $numericTable
 							WHERE name LIKE 'done%'
 								AND value = ". Piwik_ArchiveProcessing::DONE_OK_TEMPORARY ."
-								AND ts_archived < ?", array($purgeArchivesOlderThan));
+								AND period = ?
+								AND ts_archived < ?", array($this->period, $purgeArchivesOlderThan));
 			
 			$idArchivesToDelete = array();
 			if(!empty($result))
