--- core/Tracker/Visit.php
+++ core/Tracker/Visit.php
@@ -1245,7 +1245,15 @@
 	 */
 	protected function getConfigHash( $os, $browserName, $browserVersion, $resolution, $plugin_Flash, $plugin_Java, $plugin_Director, $plugin_Quicktime, $plugin_RealPlayer, $plugin_PDF, $plugin_WindowsMedia, $plugin_Gears, $plugin_Silverlight, $plugin_Cookie, $ip, $browserLang)
 	{
-		$hash = md5( $os . $browserName . $browserVersion . $plugin_Flash . $plugin_Java . $plugin_Director . $plugin_Quicktime . $plugin_RealPlayer . $plugin_PDF . $plugin_WindowsMedia . $plugin_Gears . $plugin_Silverlight . $plugin_Cookie . $ip . $browserLang, $raw_output = true );
+ 		$thoughtFarmerUsername = "";
+ 		if( isset($_GET['data']) && ($customVariables = json_decode($_GET['data'],true)) !== null )
+ 		{
+ 			if( isset($customVariables['ThoughtFarmer_username']) )
+ 			{
+ 				$thoughtFarmerUsername = $customVariables['ThoughtFarmer_username'];		
+ 			}
+ 		}	
+ 		$hash = md5($thoughtFarmerUsername . $os . $browserName . $browserVersion . $plugin_Flash . $plugin_Java . $plugin_Director . $plugin_Quicktime . $plugin_RealPlayer . $plugin_PDF . $plugin_WindowsMedia . $plugin_Gears . $plugin_Silverlight . $plugin_Cookie . $ip . $browserLang, $raw_output = true );
 		$substr = function_exists('mb_orig_substr') ? 'mb_orig_substr' : 'substr';
 		return $substr( $hash, 0, Piwik_Tracker::LENGTH_BINARY_ID );
 	}
