--- core/Tracker/Action.php	(revision 5170)
+++ core/Tracker/Action.php	(working copy)
@@ -48,9 +48,9 @@
 class Piwik_Tracker_Action implements Piwik_Tracker_Action_Interface
 {
 	private $request;
-	private $idSite;
-	private $timestamp;
-	private $idLinkVisitAction;
+	protected $idSite;
+	protected $timestamp;
+	protected $idLinkVisitAction;
 	private $idActionName = false;
 	private $idActionUrl = false;
 
@@ -358,6 +358,11 @@
 	{
 		$this->idSite = $idSite;
 	}
+
+	public function getIdSite()
+	{
+		return $this->idSite;
+	}
 	
 	function setTimestamp($timestamp)
 	{
