Index: core/Tracker/Action.php
===================================================================
--- core/Tracker/Action.php	(revision 4837)
+++ core/Tracker/Action.php	(working copy)
@@ -45,9 +45,9 @@
 class Piwik_Tracker_Action implements Piwik_Tracker_Action_Interface
 {
 	private $request;
-	private $idSite;
-	private $timestamp;
-	private $idLinkVisitAction;
+	protected $idSite;
+	protected $timestamp;
+	protected $idLinkVisitAction;
 	private $idActionName = null;
 	private $idActionUrl = null;
 
@@ -265,7 +265,7 @@
 			printDebug("Recording a new page URL in the lookup table: ". $this->idActionUrl);
 		}
 	}
-	
+
 	/**
 	 * @param int $idSite
 	 */
@@ -273,6 +273,11 @@
 	{
 		$this->idSite = $idSite;
 	}
+
+	public function getIdSite()
+	{
+		return $this->idSite;
+	}
 	
 	function setTimestamp($timestamp)
 	{
@@ -328,7 +333,7 @@
 		*/ 
 		Piwik_PostEvent('Tracker.Action.record', $this, $info);
 	 }
-	 
+
 	/**
 	 * Returns the ID of the newly created record in the log_link_visit_action table
 	 *
@@ -338,7 +343,7 @@
 	{
 		return $this->idLinkVisitAction;
 	}
-	
+
 	 /**
 	 * Generates the name of the action from the URL or the specified name.
 	 * Sets the name as $this->actionName
Index: core/ViewDataTable.php
===================================================================
--- core/ViewDataTable.php	(revision 4837)
+++ core/ViewDataTable.php	(working copy)
@@ -144,6 +144,14 @@
 	protected $columnsToDisplay = array();
 
 	/**
+	 * Array of columns set to display in table
+	 *
+	 * @var array
+	 */
+	protected $tableAllColumnsToDisplay = array();
+
+
+	/**
 	 * Variable that is used as the DIV ID in the rendered HTML
 	 *
 	 * @var string
@@ -366,7 +374,7 @@
 	{
 		// we build the request string (URL) to call the API
 		$requestString = $this->getRequestString();
-		
+
 		// we make the request to the API
 		$request = new Piwik_API_Request($requestString);
 		
@@ -961,6 +969,19 @@
 	}
 
 	/**
+	 * Overrides the columns that will be displayed in the HTML output all columns view
+	 * By default they are defined in Piwik_ViewDataTable_HtmlTable_AllColumns class
+	 */
+	public function setTableAllColumnsToDisplay( $columnsNames )
+	{
+		if(!is_array($columnsNames))
+		{
+			$columnsNames = array($columnsNames);
+		}
+		$this->tableAllColumnsToDisplay = $columnsNames;
+	}
+
+	/**
 	 * Returns columns names to display, in order.
 	 * If no columns were specified to be displayed, return all columns found in the first row.
 	 * @param array PHP array conversion of the data table
Index: core/Piwik.php
===================================================================
--- core/Piwik.php	(revision 4837)
+++ core/Piwik.php	(working copy)
@@ -383,7 +383,7 @@
 				}
 			}
 		}
-		$directoryMessage = "<p><b>Piwik couldn't write to some directories</b>.</p> <p>Try to Execute the following commands on your server:</p>"
+		$directoryMessage = "<p><b>Intranet Statistics couldn't write to some directories</b>.</p> <p>Try to Execute the following commands on your server:</p>"
 		                  . "<blockquote>$directoryList</blockquote>"
 		                  . "<p>If this doesn't work, you can try to create the directories with your FTP software, and set the CHMOD to 0777 (with your FTP software, right click on the directories, permissions).</p>"
 		                  . "<p>After applying the modifications, you can <a href='index.php'>refresh the page</a>.</p>"
@@ -1393,7 +1393,7 @@
 	 */
 	public function getLogoPath()
 	{
-		return Piwik_Common::getPathToPiwikRoot() . '/themes/default/images/logo.png';
+		return Piwik_Common::getPathToPiwikRoot() . '/themes/default/images/intranet-statistics-logo-02.jpg';
 	}
 
 	/**
Index: core/Tracker/Visit.php
===================================================================
--- core/Tracker/Visit.php	(revision 4837)
+++ core/Tracker/Visit.php	(working copy)
@@ -1144,7 +1144,15 @@
 	 */
 	protected function getConfigHash( $os, $browserName, $browserVersion, $resolution, $plugin_Flash, $plugin_Java, $plugin_Director, $plugin_Quicktime, $plugin_RealPlayer, $plugin_PDF, $plugin_WindowsMedia, $plugin_Gears, $plugin_Silverlight, $plugin_Cookie, $ip, $browserLang)
 	{
-		$hash = md5( $os . $browserName . $browserVersion . $plugin_Flash . $plugin_Java . $plugin_Director . $plugin_Quicktime . $plugin_RealPlayer . $plugin_PDF . $plugin_WindowsMedia . $plugin_Gears . $plugin_Silverlight . $plugin_Cookie . $ip . $browserLang, $raw_output = true );
+		$thoughtFarmerUsername = "";
+		if( isset($_GET['data']) && ($customVariables = json_decode(stripslashes($_GET['data']),true)) !== null )
+		{
+			if( isset($customVariables['ThoughtFarmer_username']) )
+			{
+				$thoughtFarmerUsername = $customVariables['ThoughtFarmer_username'];		
+			}
+		}	
+		$hash = md5($thoughtFarmerUsername . $os . $browserName . $browserVersion . $plugin_Flash . $plugin_Java . $plugin_Director . $plugin_Quicktime . $plugin_RealPlayer . $plugin_PDF . $plugin_WindowsMedia . $plugin_Gears . $plugin_Silverlight . $plugin_Cookie . $ip . $browserLang, $raw_output = true );
 		return substr( $hash, 0, Piwik_Tracker::LENGTH_BINARY_ID);
 	}
 
