--- core/ViewDataTable.php	(revision 5170)
+++ core/ViewDataTable.php	(working copy)
@@ -160,6 +160,14 @@
 	protected $columnsToDisplay = array();
 
 	/**
+	 * Array of columns set to display in table
+	 *
+	 * @var array
+	 */
+	protected $tableAllColumnsToDisplay = array();
+
+
+	/**
 	 * Variable that is used as the DIV ID in the rendered HTML
 	 *
 	 * @var string
@@ -1048,6 +1056,19 @@
 	}
 
 	/**
+	 * Overrides the columns that will be displayed in the HTML output all columns view
+	 * By default they are defined in Piwik_ViewDataTable_HtmlTable_AllColumns class
+	 */
+	public function setTableAllColumnsToDisplay( $columnsNames )
+	{
+		if(!is_array($columnsNames))
+		{
+			$columnsNames = array($columnsNames);
+		}
+		$this->tableAllColumnsToDisplay = $columnsNames;
+	}
+
+	/**
 	 * Returns columns names to display, in order.
 	 * If no columns were specified to be displayed, return all columns found in the first row.
 	 * @param array PHP array conversion of the data table
