--- misc/cron/archive-day.sh	(revision 5170)
+++ misc/cron/archive-day.sh	(working copy)
@@ -1,4 +1,4 @@
-#!/bin/sh -e
+#!/bin/sh +e
 
 # Description
 # This cron script will automatically run Piwik archiving every hour.
@@ -70,7 +70,7 @@
 for idsite in $ID_SITES; do
   TEST_IS_NUMERIC=`echo $idsite | egrep '^[0-9]+$'`
   if test -n "$TEST_IS_NUMERIC"; then
-    for period in day week month year; do
+    for period in day; do
       echo ""
       echo "Archiving period = $period for idsite = $idsite..."
       CMD="$PHP_BIN -q $PIWIK_PATH -- module=API&method=VisitsSummary.getVisits&idSite=$idsite&period=$period&date=last52&format=xml&token_auth=$TOKEN_AUTH"
