Index: plugins/Live/Visitor.php
===================================================================
--- plugins/Live/Visitor.php	(revision 4837)
+++ plugins/Live/Visitor.php	(working copy)
@@ -90,6 +90,7 @@
 			'screenTypeIcon' => $this->getScreenTypeIcon(),
 			'plugins' => $this->getPlugins(),
 			'pluginsIcons' => $this->getPluginIcons(),
+			'liveUsername' => $this->getLiveUsername()
 		);
 	}
 
@@ -383,4 +384,12 @@
 	{
 		return $this->details['visit_goal_converted'];
 	}
+
+	function getLiveUsername()
+	{
+		if(isset($this->details['thoughtfarmer_username'])){
+			return $this->details['thoughtfarmer_username'];
+		}
+		return false;
+	}
 }
Index: plugins/Live/templates/visitorLog.tpl
===================================================================
--- plugins/Live/templates/visitorLog.tpl	(revision 4837)
+++ plugins/Live/templates/visitorLog.tpl	(working copy)
@@ -56,7 +56,8 @@
 						<br/><acronym title="{'CustomVariables_CustomVariables'|translate} (index {$id})">{$customVariable.$name}</acronym>: {$customVariable.$value}
 					{/foreach}
 				{/if}
-				
+				<br />{$visitor.columns.liveUsername}
+
 	</td>
 	<td class="label" style="width:13%" width="13%">
 		&nbsp;<img src="{$visitor.columns.countryFlag}" title="{$visitor.columns.country}, Provider {$visitor.columns.provider}" />
Index: plugins/Live/templates/lastVisits.tpl
===================================================================
--- plugins/Live/templates/lastVisits.tpl	(revision 4837)
+++ plugins/Live/templates/lastVisits.tpl	(working copy)
@@ -14,7 +14,7 @@
 				<span class='hash'>#</span>{$visitor.goalConversions}
 				</span>{/if}
 				{if $visitor.visitorType=='returning'}&nbsp;<img src="plugins/Live/templates/images/returningVisitor.gif" title="{'General_ReturningVisitor'|translate}" />{/if}
-				{if $visitor.visitIp}- <span title="{if !empty($visitor.visitorId)}{'General_VisitorID'|translate}: {$visitor.visitorId}{/if}">IP: {$visitor.visitIp}</span>{/if}
+				&nbsp;{$visitor.liveUsername}
 			</div>
 			<!--<div class="settings"></div>-->
 			<div class="referer">
