--- plugins/CoreHome/templates/header.tpl
+++ plugins/CoreHome/templates/header.tpl
@@ -2,7 +2,7 @@
    "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
 <html xmlns="http://www.w3.org/1999/xhtml">
 <head>
-<title>{if !$isCustomLogo}Piwik &rsaquo; {/if} {'CoreHome_WebAnalyticsReports'|translate} - {$siteName}</title>
+<title>Intranet Statistics &rsaquo; {'CoreHome_WebAnalyticsReports'|translate} - {$siteName}</title>
 <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
 <meta name="generator" content="Piwik - Open Source Web Analytics" />
 <meta name="description" content="Web Analytics report for '{$siteName}' - Piwik" />
