--- plugins/Goals/Controller.php	(revision 5170)
+++ plugins/Goals/Controller.php	(working copy)
@@ -207,7 +207,7 @@
 		$view = $this->getOverviewView();
 		$view->goalsJSON = json_encode($this->goals);
         $view->goalDimensions = Piwik_Goals::getReportsWithGoalMetrics();
-		$view->userCanEditGoals = Piwik::isUserHasAdminAccess($this->idSite);
+		$view->userCanEditGoals = Piwik::isUserHasViewAccess($this->idSite);
 		$view->ecommerceEnabled = $this->site->isEcommerceEnabled();
 		$view->displayFullReport = true;
 		echo $view->render();
@@ -277,7 +277,7 @@
 	{
 		$view = Piwik_View::factory('add_new_goal');
 		$this->setGeneralVariablesView($view);
-		$view->userCanEditGoals = Piwik::isUserHasAdminAccess($this->idSite);
+		$view->userCanEditGoals = Piwik::isUserHasViewAccess($this->idSite);
 		$view->onlyShowAddNewGoal = true;
 		echo $view->render();
 	}
