--- plugins/Actions/Controller.php	(revision 5170)
+++ plugins/Actions/Controller.php	(working copy)
@@ -58,6 +58,14 @@
 		return $this->renderView($view, $fetch);
 	}
 
+	protected function configureViewPageTitles($view)
+	{
+		$view->setColumnsToDisplay( array('label','nb_hits','nb_visits','nb_page_edit','nb_page_comment') );
+
+		$view->setColumnTranslation('nb_page_edit', 'Edits');
+		$view->setColumnTranslation('nb_page_comment', 'Comments');
+	}
+
 	protected function configureViewPages($view)
 	{
 		$view->setColumnsToDisplay( array('label','nb_hits','nb_visits', 'bounce_rate', 'avg_time_on_page', 'exit_rate') );
@@ -169,7 +177,7 @@
 						'Actions.getPageTitles',
 						'getPageTitlesSubDataTable' );
 		$view->setColumnTranslation('label', Piwik_Translate('Actions_ColumnPageName'));
-		$this->configureViewPages($view);
+		$this->configureViewPageTitles($view);
 		$this->configureViewActions($view);
 		return $this->renderView($view, $fetch);
 	}
@@ -181,7 +189,7 @@
 						__FUNCTION__,
 						'Actions.getPageTitles',
 						'getPageTitlesSubDataTable'  );
-		$this->configureViewPages($view);
+		$this->configureViewPageTitles($view);
 		$this->configureViewActions($view);
 		return $this->renderView($view, $fetch);
 	}
