Index: plugins/Live/Visitor.php
===================================================================
--- plugins/Live/Visitor.php	(revision 3237)
+++ plugins/Live/Visitor.php	(working copy)
@@ -81,7 +81,8 @@
 			'goalName' => $this->getGoalName(),
    			'goalRevenue' => $this->getGoalRevenue(),
 			'goalUrl' => $this->getGoalUrl(),
-   			'goalTimePretty' => $this->getGoalTimePretty()
+   			'goalTimePretty' => $this->getGoalTimePretty(),
+			'liveUsername' => $this->getLiveUsername()
 		);
 	}
 
@@ -379,4 +380,12 @@
 		}
 		return false;
 	}
+
+	function getLiveUsername()
+	{
+		if(isset($this->details['thoughtfarmer_username'])){
+			return $this->details['thoughtfarmer_username'];
+		}
+		return false;
+	}
 }
Index: plugins/Live/templates/visitorLog.tpl
===================================================================
--- plugins/Live/templates/visitorLog.tpl	(revision 3237)
+++ plugins/Live/templates/visitorLog.tpl	(working copy)
@@ -42,7 +42,8 @@
 						{$visitor.columns.provider}
 					</a>
 				{/if}
-				
+				<br />{$visitor.columns.liveUsername}
+
 	</td>
 	<td class="label" style="width:13%" width="13%">
 		&nbsp;<img src="{$visitor.columns.countryFlag}" title="{$visitor.columns.country}, Provider {$visitor.columns.provider}" />
Index: plugins/Live/templates/lastVisits.tpl
===================================================================
--- plugins/Live/templates/lastVisits.tpl	(revision 3237)
+++ plugins/Live/templates/lastVisits.tpl	(working copy)
@@ -7,8 +7,7 @@
 				&nbsp;<img src="{$visitor.browserIcon}" title="{$visitor.browser} with plugins {$visitor.plugins} enabled" />
 				&nbsp;<img src="{$visitor.operatingSystemIcon}" title="{$visitor.operatingSystem}, {$visitor.resolution}" />
 				&nbsp;{if $visitor.isVisitorGoalConverted}<img src="{$visitor.goalIcon}" title="{$visitor.goalType}" />{/if}
-				{if $visitor.isVisitorReturning}&nbsp;<img src="plugins/Live/templates/images/returningVisitor.gif" title="Returning Visitor" />{/if}
-				&nbsp;<label id="" title="IP: {$visitor.ip} - Duration: {$visitor.visitLengthPretty}">more...</label>
+				&nbsp;{$visitor.liveUsername}
 			</div>
 			<!--<div class="settings"></div>-->
 			<div class="referer">
