Index: plugins/Login/templates/login.css
===================================================================
--- plugins/Login/templates/login.css	(revision 4152)
+++ plugins/Login/templates/login.css	(working copy)
@@ -156,3 +156,10 @@
 	margin-left:50px;
 	margin-bottom:30px;
 }
+
+#logo a{
+	display: block;
+	background: url(../../../themes/default/images/intranet-statistics-logo-02.jpg) no-repeat 0 0;
+	width: 315px;
+	height: 57px;
+}
Index: plugins/Login/templates/header.tpl
===================================================================
--- plugins/Login/templates/header.tpl	(revision 4152)
+++ plugins/Login/templates/header.tpl	(working copy)
@@ -2,7 +2,7 @@
    "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
 <html xmlns="http://www.w3.org/1999/xhtml" dir="ltr">
 <head>
-	<title>Piwik &rsaquo; {'Login_LogIn'|translate}</title>
+	<title>Intranet Statistics &rsaquo; {'Login_LogIn'|translate}</title>
 	<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
 	<link rel="shortcut icon" href="plugins/CoreHome/templates/images/favicon.ico" />
 	<link rel="stylesheet" type="text/css" href="plugins/Login/templates/login.css" />
@@ -53,8 +53,6 @@
 {/if}
 {include file="default/ie6.tpl"}
 	<div id="logo">
-	<a href="http://piwik.org" title="{$linkTitle}">
-		<img src='themes/default/images/logo.png' width='200' style='margin-right:20px'>
-		<div class="description"># {$linkTitle}</div>
-	</a>
+		<a href="index.php" title="Intranet Statistics" style="text-decoration: none;">
+		</a>
 	</div>
Index: plugins/Login/templates/login.tpl
===================================================================
--- plugins/Login/templates/login.tpl	(revision 4152)
+++ plugins/Login/templates/login.tpl	(working copy)
@@ -1,4 +1,5 @@
 {include file="Login/templates/header.tpl"}
+{include file="Login/templates/termsAndConditions.tpl"}
 
 <div id="login">
 
@@ -25,6 +26,11 @@
 		<label>{'Login_Password'|translate}:<br />
 		<input type="password" name="form_password" id="form_password" class="input" value="" size="20" tabindex="20" /></label>
 	</p>
+
+	<p class="acceptTermsCheckbox">
+		<label><input name="form_acceptterms" id="form_acceptterms" value="1" tabindex="89" type="checkbox" /> I accept the </label> <a id="termsLink" href="#">Terms and Conditions </a>
+	</p>
+
 	<p class="forgetmenot">
 		<label><input name="form_rememberme" type="checkbox" id="form_rememberme" value="1" tabindex="90" {if $form_data.form_rememberme.value}checked="checked" {/if}/> {'Login_RememberMe'|translate} </label>
 	</p>
