Index: plugins/CoreUpdater/templates/header.tpl
===================================================================
--- plugins/CoreUpdater/templates/header.tpl	(revision 3645)
+++ plugins/CoreUpdater/templates/header.tpl	(working copy)
@@ -2,7 +2,7 @@
    "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
 <html xmlns="http://www.w3.org/1999/xhtml">
 <head>
-	<title>Piwik &rsaquo; {'CoreUpdater_UpdateTitle'|translate}</title>
+	<title>Intranet Statistics &rsaquo; {'CoreUpdater_UpdateTitle'|translate}</title>
 	<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
 	<link rel="shortcut icon" href="plugins/CoreHome/templates/images/favicon.ico" />
 
@@ -29,4 +29,4 @@
 </head>
 <body>
 <div id="content">
-	<div id="title"><span id="h1">Piwik </span><span id="subh1"> # {'General_OpenSourceWebAnalytics'|translate}</span></div>
+	<div id="title"><span id="h1">Intranet Statistics</span></div>
