--- themes/default/simple_structure_header.tpl	(revision 5170)
+++ themes/default/simple_structure_header.tpl	(working copy)
@@ -9,4 +9,4 @@
 </head>
 <body>
 <div id="content">
-	<div id="title"><img title='Piwik' alt="Piwik" src='themes/default/images/logo-header.png' style='margin-left:10px' /><span id="subh1"> # <a href='http://piwik.org/'>web analytics</a></span></div>
+	<div id="title"><span id="h1">Intranet Statistics</span></div>	
