Index: themes/default/common.css
===================================================================
--- themes/default/common.css	(revision 4837)
+++ themes/default/common.css	(working copy)
@@ -213,6 +213,12 @@
 	font:42px Georgia, "Times New Roman", Times, serif;
 	height:40px;
 }
+#logo a{
+	display: block;
+	background: url(../../themes/default/images/intranet-statistics-logo-02.jpg) no-repeat 0 0;
+	width: 315px;
+	height: 57px;
+}
 
 
 
Index: themes/default/simple_structure_header.tpl
===================================================================
--- themes/default/simple_structure_header.tpl	(revision 4837)
+++ themes/default/simple_structure_header.tpl	(working copy)
@@ -9,4 +9,4 @@
 </head>
 <body>
 <div id="content">
-	<div id="title"><img title='Piwik' src='themes/default/images/logo-header.png' style='margin-left:10px'><span id="subh1"> # <a href='http://piwik.org/'>web analytics</a></span></div>
+	<div id="title"><span id="h1">Intranet Statistics</span></div>
