Index: core/Url.php
===================================================================
--- core/Url.php (revision 3632)
+++ core/Url.php (revision 3649)
@@ -121,5 +121,16 @@
 		if(empty($url))
 		{
-			$url = $_SERVER['SCRIPT_NAME'];
+			if(isset($_SERVER['SCRIPT_NAME']))
+			{
+				$url = $_SERVER['SCRIPT_NAME'];
+			}
+			elseif(isset($_SERVER['SCRIPT_FILENAME']))
+			{
+				$url = $_SERVER['SCRIPT_FILENAME'];
+			}
+			elseif(isset($argv))
+			{
+				$url = $argv[0];
+			}
 		}
 
